src/ruby/ext/grpc/extconf.rb

```diff
@@ -141,7 +141,7 @@ def ext_export_filename()
 
 ext_export_file = File.join(grpc_root, 'src', 'ruby', 'ext', 'grpc', ext_export_filename())
 $LDFLAGS << ' -Wl,--version-script="' + ext_export_file + '.gcc"' if linux
-$LDFLAGS << ' -Wl,-exported_symbols_list,"' + ext_export_file + '.clang"' if apple_toolchain
+$LDFLAGS << ' -weak_framework CoreFoundation -Wl,-exported_symbols_list,"' + ext_export_file + '.clang"' if apple_toolchain
```

Comment:
can we add this flag only when we're building apple *and* arm?

Reply:
See https://github.com/grpc/grpc/pull/33538 for this change

Reply:
Sorry for late response, I was have access to x86 Macs to verify this doesn't happen with x86 when building from source. Now I have verified this only happens to ARM Macs.

Thanks for making the change, closing this PR.

Reply:
I had someone on my team report the same issue on `x86-darwin`, but we used the precompiled gem as a workaround.